src/DefaultBuilder/src/StaticWebAssetsLoader.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal class StaticWebAssetsLoader
+    {
+        internal const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+
+        internal static void UseStaticWebAssets(IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var additionalFiles = StaticWebAssetsReader.Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (additionalFiles.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                additionalFiles.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(additionalFiles);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            // We plan to remove the embedded file resolution code path in
+            // a future preview.
+            Assembly assembly = null;
+            try
+            {
+                assembly = Assembly.Load(environment.ApplicationName);
+            }
+            catch (Exception)
+            {
+            }
+
+            if (assembly != null && assembly.GetManifestResourceNames().Any(a => a == StaticWebAssetsManifestName))
```

Comment:
When are things embedded attributes? Identity?

Reply:
No, this is going away in preview7 as we moved to use a file. Just need it here until the Razor SDK flows.